models/stg_twilio__call.sql

```diff
@@ -29,7 +29,7 @@ final as (
         caller_name,
         created_at,
         direction,
-        duration,
+        cast(duration as {{ dbt.type_float() }}) as duration,
```

Comment:
To be safe, I think it'd be good to scrub out any non-numeric characters (anything that's not a digit, decimal point, or negative sign) out of each of the fields before we cast them from strings to floats. This'll avoid errors if there are weird edge cases that Twilio hasn't explicitly called out 

To do so, we could use REGEXP_REPLACE, which is a tiny bit different per warehouse. For a macro, ChatGPT suggests this (though if we're doing a macro I'd suggest using adapter.dispatch logic instead of target.type conditionals):
```sql
{% macro remove_non_numeric_chars(input_string) %}
    {% if target.type == 'bigquery' %}
        REGEXP_REPLACE({{ input_string }}, r'[^0-9.-]', '')
    {% elif target.type == 'snowflake' %}
        REGEXP_REPLACE({{ input_string }}, '[^0-9.-]', '')
    {% elif target.type == 'redshift' %}
        REGEXP_REPLACE({{ input_string }}, '[^0-9.-]', '')
    {% elif target.type == 'postgres' %}
        REGEXP_REPLACE({{ input_string }}, '[^0-9.-]', '', 'g')
    {% elif target.type == 'databricks' %}
        REGEXP_REPLACE({{ input_string }}, '[^0-9.-]', '')
    {% else %}
        {{ exceptions.raise_compiler_error("This macro is not implemented for the current adapter: " ~ target.type) }}
    {% endif %}
{% endmacro %}
```

Reply:
Great thinking, added a macro